src/components/ChangePlan/Checkout/ContactInformation/ContactInformation.js

```diff
@@ -0,0 +1,255 @@
+import React, { useEffect, useState } from 'react';
+import { InjectAppServices } from '../../../../services/pure-di';
+import { useIntl } from 'react-intl';
+import useTimeout from '../../../../hooks/useTimeout';
+import {
+  FieldGroup,
+  FieldItem,
+  InputFieldItem,
+  PhoneFieldItem,
+  SelectFieldItem,
+  SubmitButton,
+} from '../../../form-helpers/form-helpers';
+import { Form, Formik } from 'formik';
+import { getCountries } from '../../../../utils';
+import { Loading } from '../../../Loading/Loading';
+
+export const ContactInformation = InjectAppServices(
+  ({ dependencies: { dopplerUserApiClient }, handleSaveAndContinue, onComplete }) => {
+    const intl = useIntl();
+    const _ = (id, values) => intl.formatMessage({ id: id }, values);
+    const defaultOption = { key: '', value: _('checkoutProcessForm.empty_option_select') };
+    const language = intl.locale;
+    const [state, setState] = useState({ loading: true });
+
+    const fieldNames = {
+      firstname: 'firstname',
+      lastname: 'lastname',
+      address: 'address',
+      city: 'city',
+      province: 'province',
+      country: 'country',
+      zipCode: 'zipCode',
+      phone: 'phone',
+      company: 'company',
+      industry: 'industry',
+    };
+
+    useEffect(() => {
+      const fetchData = async () => {
+        const contactInformationResult = await dopplerUserApiClient.getContactInformationData();
+
+        if (contactInformationResult.success) {
+          onComplete(contactInformationResult.value.completed);
+        }
+
+        setState({
+          contactInformation: contactInformationResult.success
+            ? contactInformationResult.value
+            : null,
+          success: contactInformationResult.success,
+          loading: false,
+        });
+      };
+      fetchData();
+    }, [dopplerUserApiClient]);
+
+    const getFormInitialValues = () => {
+      let initialValues = Object.keys(fieldNames).reduce(
+        (accumulator, currentValue) => ({
+          ...accumulator,
+          [currentValue]: '',
+        }),
+        {},
+      );
+
+      if (state.contactInformation) {
+        initialValues = { ...state.contactInformation };
+      }
+
+      return initialValues;
+    };
+
+    const [formSubmitted, setFormSubmitted] = useState(false);
+    const createTimeout = useTimeout();
+
+    const submitContactInformationForm = async (values, { setSubmitting }) => {
+      setFormSubmitted(false);
+
+      const result = await dopplerUserApiClient.createOrUpdateContactInformation(values);
+      if (result.success) {
+        setFormSubmitted(true);
+        createTimeout(() => {
```

Comment:
Why do you need a timeout here?

Reply:
The timeout is used to show the success message for 3 seconds. In this way the user can to see the message that the information was saved succeed.

Reply:
![image](https://user-images.githubusercontent.com/70591946/125340219-956bd080-e328-11eb-9c46-2eb9a78de487.png)